pkg/gardenlet/controller/federatedseed/extensions/shootextensionstatus_control.go

```diff
@@ -0,0 +1,234 @@
+
+package extensions
+
+import (
+	"context"
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/sirupsen/logrus"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/extensions"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+// ShootExtensionStatusControl is used to sync the `providerStatus` field from extension
+// resources in the Seed to the ShootExtensionStatus resource in the Garden cluster.
+type ShootExtensionStatusControl struct {
+	gardenClient client.Client
+	seedClient   client.Client
+	log          *logrus.Entry
+	recorder     record.EventRecorder
+	decoder      runtime.Decoder
+}
+
+// NewShootExtensionStatusControl creates a new instance of ShootExtensionStatusControl.
+func NewShootExtensionStatusControl(k8sGardenClient, seedClient client.Client, log *logrus.Entry, recorder record.EventRecorder) *ShootExtensionStatusControl {
+	return &ShootExtensionStatusControl{
+		gardenClient: k8sGardenClient,
+		seedClient:   seedClient,
+		log:          log,
+		recorder:     recorder,
+		decoder:      extensions.NewGardenDecoder(),
+	}
+}
+
+// CreateShootExtensionStatusSyncReconcileFunc creates a function which can be used by the reconciliation loop to sync
+// the extension status to the ShootExtensionStatus resource in the Garden cluster
+func (s *ShootExtensionStatusControl) CreateShootExtensionStatusSyncReconcileFunc(kind string, objectCreator func() client.Object) reconcile.Func {
+	return func(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+		extensionObj, err := GetExtensionObject(ctx, s.seedClient, req.NamespacedName, objectCreator)
+		if apierrors.IsNotFound(err) {
+			// the extension has been deleted from the Seed cluster
+			// we an be sure that there are no more cloud provider resources left.
+			// Delete the provider status from the ShootExtensionStatus resource in the Garden cluster
+			s.log.Debugf("Extension resource (%s/%s) of kind %q has been deleted. Synchonrizing with ShootExtensionStatus in the Garden cluster", req.NamespacedName.Namespace, req.NamespacedName.Name, kind)
+
+			shoot, err := s.getShootForRequest(ctx, req)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to delete the provider status of the %q extension (%s/%s) from the ShootExtensionStatus: %w",
+					kind,
+					req.NamespacedName.Namespace,
+					req.NamespacedName.Name,
+					err)
+			}
+
+			shootExtensionStatus := &gardencorev1alpha1.ShootExtensionStatus{}
```

Comment:
Can you check how this will play with the `SeedAuthorizer`? 
I guess we can't/shouldn't simply watch all `ShootExtensionStatus`es.

Reply:
Good point, though I would propose to take care of the seed restriction admission plugin & authorizer webhook in a follow-up PR. I would then investigate and check how to watch only `ShootExtensionStatuses` for the Seed it is responsible for.

Reply:
> though I would propose to take care of the seed restriction admission plugin & authorizer webhook in a follow-up PR.

At this point, we think only `Secret` handling is missing in the seed authorizer. I plan to open a PR for it next week. After it's open/merge merge, I would like to go ahead and play with Gardener while activating the seed authorizer.
Hence, if we now NOT merge this PR and don't have the necessary code for `ShootExtensionStatus`es in the seed authorizer, I'm blocked. So, please add it to this PR instead of postponing it (it shouldn't be too much of necessary code).

> If yes, what can be done about it?

You might want to look into https://github.com/gardener/gardener/pull/3803.